### PR TITLE
[FIX] account: add missing fields to mobile aml views

### PR DESCRIPTION
**Pull request by yelizariev**
Party is boring when some important guests are not invited. More precisely,
product quantity is not saved after editing an invoice.

In Odoo Mobile App [1], when we create a new Invoice in Accounting app, we can
add a product and input an amount. After saving adding the product and saving
the invoice, the quantity of the product saves correctly.

Next, Edit the Invoice again and add quantity -> Save adding product. We then
save the Invoice -> the quantity reverts back to the initial save and new
quantity is not updated.

A similar fix [2] was done for kanban view for `invoice_line_ids`, but not for `line_ids` field.

[1] The issue is also reproduced if we activate mobile device emulation in
browser

[2]: https://github.com/odoo/odoo/commit/358f6330281848e6c33f6d2e5a7f890eaa7b0e02

opw-2945805
opw-2806178

### CODE REVIEW

**Comment by yelizariev** [comment]
Also, opw-2806178

**Comment by yelizariev** [comment]
@smetl :pray:

File: addons/account/views/account_move_views.xml [hunk 1]
```diff
@@ -94,6 +94,31 @@
...
+                    <field name="tax_repartition_line_id"/>
+                    <field name="tax_base_amount"/>
+                    <field name="company_id"/>
+                    <field name="recompute_tax_line"/>
+                    <field name="display_type"/>
+                    <field name="is_rounding_line"/>
+                    <field name="exclude_from_invoice_tab"/>
+                    <field name="account_internal_type"/>
+                    <field name="account_internal_group"/>
+
```
Comment (vin-odoo) [inline addons/account/views/account_move_views.xml:121]: Some of these are not used in a kanban view, do you still need them here?
payment_id, discount, sequence, move_name, tax_group_id
```suggestion
                    <field name="product_id"/>
                    <field name="quantity"/>
                    <field name="product_uom_category_id"/>
                    <field name="product_uom_id"/>
                    <field name="price_unit"/>
                    <field name="price_subtotal"/>
                    <field name="price_total"/>
                    <field name="date"/>
                    <field name="group_tax_id"/>
                    <field name="tax_line_id"/>
                    <field name="tax_repartition_line_id"/>
                    <field name="tax_base_amount"/>
                    <field name="company_id"/>
                    <field name="recompute_tax_line"/>
                    <field name="display_type"/>
                    <field name="is_rounding_line"/>
                    <field name="exclude_from_invoice_tab"/>
                    <field name="account_internal_type"/>
                    <field name="account_internal_group"/>

```

**Comment by yelizariev** [comment]
@vin-odoo I've tried without extra fields you listed. Updating qty fields works indeed, but total taxes of the invoices are set to zero after saving the record. I don't really know which field exactly makes difference. For me just adding all fields is fine and doesn't really have downsides

File: addons/account/views/account_move_views.xml [hunk 2]
```diff
@@ -94,6 +94,31 @@
                 <kanban class="o_kanban_mobile" create="false" group_create="false">
                     <field name="company_currency_id"/>
                     <field name="partner_id"/>
+                    <field name="payment_id"/>
```
Comment (william-andre) [inline addons/account/views/account_move_views.xml:97]: Why do you need it in the kanban? It is not editable, fields here are only used for display.

Am I missing something?

Comment (yelizariev) [reply addons/account/views/account_move_views.xml:97]: I don't know how it works. I copied list of fields from another view. I could only remove field one by one and check if the mobile view still works or not, but it's boring :grimacing:

Comment (yelizariev) [reply addons/account/views/account_move_views.xml:97]: :pray:

**Comment by hedshefer** [comment]
@yelizariev 
@william-andre 
Can you please update what is the status of this PR?

File: addons/account/views/account_move_views.xml (hunk 2 above)
Comment (qdp-odoo) [reply addons/account/views/account_move_views.xml:97]: well, diffs are supposed to be the minimum to fix a specific problem.... so yes, you should only include here what's fixing your issue, and not copy the whole list of fields from another view or version without "knowing how that works" :roll_eyes:

**Review by qdp-odoo — CHANGES_REQUESTED** [review]

**Comment by yelizariev** [comment]
After some experiments, I have found the minimal change to fix both quantity and tax issue in mobile mode. Take a look please. It's total 3 fields added now

**Comment by yelizariev** [comment]
I've just tested it on runbot

# Before

[screencast-22117918-15-0-design_theme.runbot176.odoo.com-2022.12.23-11_55_33.webm](https://user-images.githubusercontent.com/186131/209323967-ff434166-b842-46db-b76f-ac4532e97e00.webm)

# After

[screencast-22042898-15-0-all.runbot167.odoo.com-2022.12.23-11_52_37.webm](https://user-images.githubusercontent.com/186131/209323610-c5d21124-7003-4596-ad46-f311d7d4bb2f.webm)

**Comment by yelizariev** [comment]
@william-andre

**Comment by yelizariev** [comment]
@william-andre :pray:

**Comment by yelizariev** [comment]
:pray:

**Review by Levizar — APPROVED** [review]
@robodoo r+

The modification is now 3 lines, makes sense, is similar to what has been done previously (see the commit message), has a before/after video proving it works.

Time to move "fast" and break things :sunglasses:

**Comment by yotamlinder** [comment]
@hedshefer 
@ivantodorovich 
Hi
I can see that the issue solved in 15.0 in the runbot.
Do you know when it is going to be update in the SH?
thanks

**Comment by william-andre** [comment]
It should be good on SH already, but you would need to update the module for it to take effect.
I would recommend changing the views manually to avoid losing other potential customization. (updating `account` will trigger to update of all the modules that depend on it, which can be a lot)

**Pull request closed without merging**